clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -120,6 +123,8 @@ public Sender(KafkaClient client,
         this.clientId = clientId;
         this.sensors = new SenderMetrics(metrics);
         this.requestTimeout = requestTimeout;
+        
+        this.metadataStaleMs = getMetadataStaleMs(metadata.maxAge(), requestTimeout, metadata.refreshBackoff());
```

Comment:
I refactored the simple one-liner calculation in `getMetadataStaleMs` because I'm using the same calculation from `RecordAccumulatorTest`.